modules/core/include/opencv2/core/directx.hpp

```diff
@@ -129,6 +129,16 @@ CV_EXPORTS void convertToD3D11Texture2D(InputArray src, ID3D11Texture2D* pD3D11T
 //! @param dst             - destination OutputArray
 CV_EXPORTS void convertFromD3D11Texture2D(ID3D11Texture2D* pD3D11Texture2D, OutputArray dst);
 
+//! @brief Converts ID3D11Texture2D to cl_mem (OpenCL image2d_t).
+//
+//! @note Note: Function creates an OpenCL 2D image object from a subresource of a Direct3D 11 2D texture
+//
+//! @param pD3D11Texture2D - source D3D11 texture
+//! @param pointer to OpenCL Memory object
+//! @param pointer to OpenCL Context
+//! @param pointer to OpenCL Device
+CV_EXPORTS void convertFromD3D11Texture2DtoCLMem(ID3D11Texture2D* pD3D11Texture2D, void* clMem, void* clContext, void* clDeviceID);
```

Comment:
> void* clMem

There is `cv::ocl::Image2D`.
But its API should be improved to support this scenario.

Reply:
Added cv::ocl::Image2D and cv::ocl::Context thanks